eclair-core/src/main/scala/fr/acinq/eclair/crypto/Sphinx.scala

```diff
@@ -90,19 +89,11 @@ object Sphinx extends Logging {
    * Peek at the first bytes of the per-hop payload to extract its length.
    */
   def peekPayloadLength(payload: ByteVector): Int = {
-    payload.head match {
-      case 0 =>
-        // The 1.0 BOLT spec used 65-bytes frames inside the onion payload.
-        // The first byte of the frame (called `realm`) is set to 0x00, followed by 32 bytes of per-hop data, followed by a 32-bytes mac.
-        Metrics.OnionPayloadFormat.withTag(Tags.LegacyOnion, value = true).increment()
-        65
-      case _ =>
-        // The 1.1 BOLT spec changed the frame format to use variable-length per-hop payloads.
-        // The first bytes contain a varint encoding the length of the payload data (not including the trailing mac).
-        // Since messages are always smaller than 65535 bytes, this varint will either be 1 or 3 bytes long.
-        Metrics.OnionPayloadFormat.withTag(Tags.LegacyOnion, value = false).increment()
-        MacLength + PaymentOnionCodecs.payloadLengthDecoder.decode(payload.bits).require.value.toInt
-    }
+    require(payload.head != 0, "legacy onion format is not supported anymore")
```

Comment:
It would probably be more fool proof to make the method return a `Try[Int]` instead of relying on the caller to be careful, but that should probably be done in its own PR.

Reply:
I'll give this a try, it may not be that much more to add, in which case it makes sense to do now.

Reply:
I does change quite a lot the whole Sphinx file...I think for the best, but I'd rather do that in a separate PR.